src/odb/src/db/dbBlock.cpp

```diff
@@ -3453,6 +3453,31 @@ void dbBlock::writeGuides(const char* filename) const
   guide_file.close();
 }
 
+std::string dbBlock::generateMacroPlacementString()
```

Comment:
This method is written entirely in terms of the public API and therefore doesn't need to be a member function.  It should be moved to a utility function.